api/common/common_types.go

```diff
@@ -0,0 +1,120 @@
+	Scopes []string `json:"scopes"`
+	// UsernameClaim represents the claim contained within the OAuth JWT token which holds the username
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OAuth Username Claim",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	//+kubebuilder:default=preferred_username
+	UsernameClaim string `json:"usernameClaim"`
+	// GroupsClaim represents the claim contained within the OAuth JWT token which holds the list of groups/roles. This
+	// must be a list/array and not a space separated list of names.  It must also be a top level attribute rather than
+	// a nested field in the JSON structure of the JWT object.
+	//    i.e., {"roles": ["a", "b"]} rather than {"realm": {"roles": ["a", "b"}}.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OAuth Groups Claim",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	//+kubebuilder:default=roles
+	GroupsClaim string `json:"groupsClaim"`
+	// ClientBindingClaim represents the claim contained within the OAuth JWT token which holds the certificate SHA256
+	// fingerprint.  This is expected to be a CEL mapper expression.  It should only be changed in advanced scenarios.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OAuth Client Binding Claim",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	//+kubebuilder:default="has(claims.cnf) ? claims.cnf['x5t#S256'] : []"
+	ClientBindingClaim string `json:"clientBindingClaim"`
+}
+
+// +kubebuilder:object:generate=true
+// TLSConfig defines the TLS specific attributes
+type TLSConfig struct {
+	// SecretName represents the name of a secret (in the current namespace) which contains an X.509 certificate and
+	// private key.  The secret is expected to contain a 'tls.key' and 'tls.crt' keys.  If the client is signed by
+	// intermediate CA certificate(s), then it is expected that the full chain is appended to the certificate file with
+	// the device certificate being first and the root CA being last.  It is expected that the certificate CN and DNS SAN configured be equal
+	// to the O2IMS DNS FQDN (e.g., o2ims.apps.<cluster domain name>).  This is to ensure that the certificate can be
+	// used as the ingress certificate as well as the outgoing client certificate.
+	//+optional
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="TLS certificate"
+	SecretName *string `json:"secretName"`
+}
+
+// +kubebuilder:object:generate=true
+// AuthType defines the authorization type used for authentication.
+type AuthType string
+
+const (
+	// ServiceAccount represents authentication using a Kubernetes ServiceAccount token. This method leverages
+	// a Kubernetes ServiceAccount to authenticate requests, typically used for internal cluster communication
+	// where the client presents a JWT token issued by the Kubernetes API server.
```

Comment:
I would be explicit here and state that we assume that the plugin is going to be a service running in this same cluster... rather than in another cluster because then this wouldn't work at all and OAuth2 would need to be used instead... or we'd need to support TLS client certificate authentication to that other cluster's ingress. ... which may not be a bad idea to support, but not today!